misc/test_bit_struct.py

```diff
@@ -27,22 +27,28 @@ def foo():
 
 
 def test_simple_singleton():
-    ti.init(arch=ti.cpu, print_ir=True)
+    ti.init(arch=ti.cpu, debug=True, print_ir=True, advanced_optimization=False)
     ci13 = ti.type_factory_.get_custom_int_type(13, True)
-    cu19 = ti.type_factory_.get_custom_int_type(19, False)
+    cu14 = ti.type_factory_.get_custom_int_type(14, False)
+    ci5 = ti.type_factory_.get_custom_int_type(5, True)
 
     x = ti.field(dtype=ci13)
-    y = ti.field(dtype=cu19)
-
-    ti.root._bit_struct(num_bits=32).place(x, y)
+    y = ti.field(dtype=cu14)
+    z = ti.field(dtype=ci5)
+    ti.root._bit_struct(num_bits=32).place(x, y, z)
 
     ti.get_runtime().print_snode_tree()
     ti.get_runtime().materialize()
     ti.get_runtime().print_snode_tree()
 
     @ti.kernel
     def foo():
-        print(x[None])
+        x[None] = 2**12-1
+        assert(x[None] == 2**12-1)
+        y[None] = 2**14-1
+        assert(y[None] == 2**14-1)
+        z[None] = -(2**3)
+        assert(z[None] == -2**3)
```

Comment:
Let's also make sure `test_simple_array` passes in the tests. This file can be then removed.

Reply:
> It would be nice to move the `asserts` into a separate `bar` function. If the read and writes happen in the same kernel the optimizer may simply do store-forwarding and then the codegen is not really tested.

Thanks for telling me. It's a good idea to separate the the function. That avoids some problems caused by optimizers. Let me write the test functions in the new test file in this way.